src/itehandler/IteToSwitch.cc

```diff
@@ -204,21 +203,20 @@ ite::Dag IteToSwitch::constructIteDag(PTRef root, const Logic &logic) {
     queue.push(root);
     while (queue.size() != 0) {
         PTRef tr = queue.last();
-        const Pterm &t = logic.getPterm(tr);
-        auto index = Idx(t.getId());
-        if (flag[index] == ite::type::black) {
+        ite::type mark = type::white;
+        bool found = flag.peek(tr, mark);
```

Comment:
As discussed offline, `peek` returns a `bool` indicated whether the keys is in the map, and if yes, the second parameter will be set to the value, otherwise, it is unchanged.